packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,525 @@
+			});
+			expect(res.chain).toBe('rinkeby');
+		});
+		it('defaultHardfork', async () => {
+			// default
+			expect(web3Eth.defaultHardfork).toBe('london');
+
+			// after set
+			web3Eth.setConfig({
+				defaultHardfork: 'dao',
+			});
+			expect(web3Eth.defaultHardfork).toBe('dao');
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultHardfork: 'istanbul',
+				},
+			});
+			expect(eth2.defaultHardfork).toBe('istanbul');
+
+			const res = await prepareTransactionForSigning(
+				{
+					from: '0xEB014f8c8B418Db6b45774c326A0E64C78914dC0',
+					to: '0x3535353535353535353535353535353535353535',
+					value: '0x174876e800',
+					gas: '0x5208',
+					gasPrice: '0x4a817c800',
+					data: '0x0',
+					nonce: '0x4',
+					chainId: '0x1',
+					gasLimit: '0x5208',
+				},
+				eth2,
+			);
+			expect(res.common.hardfork()).toBe('istanbul');
+		});
+		it('defaultCommon', () => {
+			// default
+			expect(web3Eth.defaultCommon).toBeNull();
+			const common = {
+				customChain: {
+					name: 'test',
+					networkId: 123,
+					chainId: 1234,
+				},
+				baseChain: 12345,
+				hardfork: 'dao',
+			};
+			// after set
+			web3Eth.setConfig({
+				defaultCommon: common,
+			});
+			expect(web3Eth.defaultCommon).toBe(common);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultCommon: common,
+				},
+			});
+			expect(eth2.defaultCommon).toBe(common);
+		});
+		it('defaultTransactionType', () => {
+			// default
+			expect(web3Eth.defaultTransactionType).toBe('0x0');
+			// after set
+			web3Eth.setConfig({
+				defaultTransactionType: '0x3',
+			});
+			expect(web3Eth.defaultTransactionType).toBe('0x3');
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultTransactionType: '0x4444',
+				},
+			});
+			expect(eth2.defaultTransactionType).toBe('0x4444');
+
+			const res = getTransactionType(
+				{
+					from: '0xEB014f8c8B418Db6b45774c326A0E64C78914dC0',
+					to: '0x3535353535353535353535353535353535353535',
+					value: '0x174876e800',
+					gas: '0x5208',
+					data: '0x0',
+					nonce: '0x4',
+					chainId: '0x1',
+					gasLimit: '0x5208',
+				},
+				eth2,
+			);
+			expect(res).toBe('0x4444');
+		});
+		it('defaultMaxPriorityFeePerGas', async () => {
+			// default
+			expect(web3Eth.defaultMaxPriorityFeePerGas).toBe(numberToHex(2500000000));
+			// after set
+			web3Eth.setConfig({
+				defaultMaxPriorityFeePerGas: numberToHex(2100000000),
+			});
+			expect(web3Eth.defaultMaxPriorityFeePerGas).toBe(numberToHex(2100000000));
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultMaxPriorityFeePerGas: numberToHex(1200000000),
+				},
+			});
+			expect(eth2.defaultMaxPriorityFeePerGas).toBe(numberToHex(1200000000));
+
+			const res = await getTransactionGasPricing(
```

Comment:
explicitly provided maxPriorityFeePerGas should override defaultMaxPriorityFeePerGas test

Reply:
done